packages/router/src/router.ts

```diff
@@ -626,9 +652,10 @@ export class Router {
                              tap(t => {
                                if (this.urlUpdateStrategy === 'eager') {
                                  if (!t.extras.skipLocationChange) {
-                                   this.setBrowserUrl(
-                                       t.urlAfterRedirects, !!t.extras.replaceUrl, t.id,
-                                       t.extras.state);
+                                   this.setBrowserUrl(t.urlAfterRedirects, !!t.extras.replaceUrl, {
+                                     ...t.extras.state,
+                                     ...this.generateNgRouterState(t.id, t.targetPageId)
+                                   });
```

Comment:
It looks like some of the code is repeated here and below, we can probably refactor it a bit to move common code to the private function (like `setBrowserUrlWithState`), for example:
```suggestion
                                   this.setBrowserUrlWithState(t.urlAfterRedirects, t);
```
Similar to what you did with extracting common logic to the `cancelNavigationTransition` function (which is great!).